pkg/controller/apiextensions/composite/composed/api.go

```diff
@@ -54,14 +63,25 @@ func (*DefaultConfigurator) Configure(cp resource.Composite, cd resource.Compose
 	// Any existing name will be overwritten when we unmarshal the template. We
 	// store it here so that we can reset it after unmarshalling.
 	name := cd.GetName()
+	namespace := cd.GetNamespace()
```

Comment:
Will a composed resource ever have a namespace?

Reply:
It may or may not but I wanted that logic to be complete; we're setting the identifier in case template has something else and the identifier that can be overwritten is `namespace + name` here, not necessarily name only. UID could be part of the template but that's a really far away case to cover.